libs/application/templates/children-residence-change/src/fields/Overview/SignatureModal/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import { useIntl } from 'react-intl'
+import {
+  Box,
+  Text,
+  AlertMessage,
+  Button,
+  ModalBase,
+  SkeletonLoader,
+} from '@island.is/island-ui/core'
+import { signatureModal } from '../../../lib/messages'
+import { FileSignatureStatus } from '../fileSignatureReducer'
+import * as style from './Modal.treat'
+
+interface SignatureModalProps {
+  signatureStatus: FileSignatureStatus
+  onClose: () => void
+  modalOpen: boolean
+  controlCode?: string
+}
+
+const SignatureModal = ({
+  controlCode,
+  signatureStatus,
+  modalOpen,
+  onClose,
+}: SignatureModalProps) => {
+  const { formatMessage } = useIntl()
+  const hasError = [
+    FileSignatureStatus.REQUEST_ERROR,
+    FileSignatureStatus.UPLOAD_ERROR,
+  ].includes(signatureStatus)
+  return (
+    <ModalBase
+      baseId="signatureDialog"
+      className={style.modal}
+      modalLabel={formatMessage(signatureModal.general.title)}
+      isVisible={modalOpen}
+      onVisibilityChange={(visibility: boolean) => {
+        if (!visibility) {
+          onClose()
+        }
+      }}
+      // When there is an error it should not be possible to close the modal
+      hideOnEsc={hasError}
```

Comment:
How is the user flow when this happens?
Does the user see a retry button and is unable to close the modal?
Or what is the situation? 😄

Reply:
The user should not be able to close the modal with the `esc` key when the the modal is in a loading state or displaying the `controlCode` the modal should only close automatically on `success` or when there is an `error` and then the user will be able to press the `Close` button or the `esc` key to close.

Reply:
Isn't the comment on line 44 then wrong? 😊